src/connector/src/sink/kafka.rs

```diff
@@ -214,6 +214,9 @@ pub struct KafkaConfig {
     #[serde(flatten)]
     pub common: KafkaCommon,
 
+    #[serde(flatten)]
+    pub connection: crate::connector_common::KafkaConnection,
```

Comment:
nit: I guess changing this to `KafkaConnection` can make `src/connector/with_options_sink.yaml` look better